eureka2-integration/src/test/java/com/netflix/eureka2/integration/startup/FileBasedServerStartupAndShutdownIntegrationTest.java

```diff
@@ -0,0 +1,23 @@
+package com.netflix.eureka2.integration.startup;
+
+import com.netflix.eureka2.server.EurekaWriteServer;
+import org.junit.Test;
+
+/**
+ * Since this test relies on Archaius which has some global settings, we can't have different tests for both read
+ * and write servers under the same JVM. As the tests are testing startup and shutdown behaviour that is common to
+ * both (within AbstractEurekaServer), we just pick one of the two server types (write servers) to test this.
+ *
+ * @author David Liu
+ */
+public class FileBasedServerStartupAndShutdownIntegrationTest extends WriteServerStartupAndShutdownIntegrationTest {
```

Comment:
But we need this test for read server as well. We should be able to
configure write cluster addresses in read server properties file, and
bootstrap it accordingly.

On Sun, Jan 25, 2015 at 10:41 AM, qiangdavidliu <notifications@github.com>
wrote:

> In
> eureka2-integration/src/test/java/com/netflix/eureka2/integration/startup/FileBasedServerStartupAndShutdownIntegrationTest.java
> <https://github.com/Netflix/eureka/pull/354#discussion_r23506376>:
>
> > @@ -0,0 +1,23 @@
> > +package com.netflix.eureka2.integration.startup;
> > +
> > +import com.netflix.eureka2.server.EurekaWriteServer;
> > +import org.junit.Test;
> > +
> > +/**
> > + * Since this test relies on Archaius which has some global settings, we can't have different tests for both read
> > + * and write servers under the same JVM. As the tests are testing startup and shutdown behaviour that is common to
> > + * both (within AbstractEurekaServer), we just pick one of the two server types (write servers) to test this.
> > + *
> > + * @author David Liu
> > + */
> > +public class FileBasedServerStartupAndShutdownIntegrationTest extends WriteServerStartupAndShutdownIntegrationTest {
>
> It's in a separate class for clearer organisation. If we added it to the
> write test class chances are someone in the future might see the "missing"
> complementary test in the read class and add a read version back, which
> will then get us back to the same problem as now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Netflix/eureka/pull/354/files#r23506376>.
>